openerp/addons/base/ir/ir_ui_view.py

```diff
@@ -153,6 +154,22 @@ def _get_model_data(self, cr, uid, ids, fname, args, context=None):
         result.update(map(itemgetter('res_id', 'id'), data_ids))
         return result
 
+    def _convert(self, cr, uid, s, view):
+        matches = re.finditer('[^%]%\((.*?)\)[ds]', s)
```

Comment:
* `resolve_external_ids` would probably be a better name, `convert` isn't exactly meaningful.
* the parameter ordering is odd, why is the view after the arch, and why is the arch called `s`?
* why not use `re.sub`?